core/src/main/java/tc/oc/pgm/rotation/vote/MapVotePicker.java

```diff
@@ -157,18 +166,28 @@ private double getRepeatedGamemodes(List<MapInfo> selected, MapInfo map) {
   }
 
   private static final class MapVoteContext extends ExpressionContext.Impl {
-    public MapVoteContext(double score, double sameGamemode, double mapsize, double players) {
+    public MapVoteContext(
+        double score,
+        double sameGamemode,
+        double mapsize,
+        double players,
+        double participants,
+        double observers,
+        double matchDuration) {
       super(
           Map.of(
               "score", score,
               "same_gamemode", sameGamemode,
               "mapsize", mapsize,
-              "players", players),
+              "players", players,
+              "participants", participants,
+              "observers", observers,
+              "match_duration", matchDuration),
```

Comment:
`play_minutes` may be more appropriate as there's precedent (see VotingPool.Context), and it specifies the unit